autoemulate/experimental/compare.py

```diff
@@ -109,4 +113,104 @@ def compare(
                 "rmse_score": rmse_score,
             }
             self.log_compare(model_cls, best_model_config, r2_score, rmse_score)
+
+        # Store comparison results for refit method
+        self._comparison_results = models_evaluated
         return models_evaluated
+
+    def refit(
```

Comment:
At the moment the user can either refit on ALL the data (train+val+test) or on their custom dataset. Do we need an option to refit on just the train+val data. This is what AutoEmulate does at the moment as it allows the user to check the test performance of their model.

Reply:
Following comments in #588, the refit() method should by default fit on ALL the data (train+val+test) or whatever new dataset the user wants to pass it. Note that there are some additional proposed changes to `compare()` within that issue, which mean that we should not hold onto state as is implemented in this PR but rather just return the fitted models.